cf-java-logging-support-logback/src/main/java/com/sap/hcp/cf/logback/encoder/EventContextFieldSupplier.java

```diff
@@ -0,0 +1,29 @@
+package com.sap.hcp.cf.logback.encoder;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.sap.hcp.cf.logback.converter.api.LogbackContextFieldSupplier;
+import com.sap.hcp.cf.logging.common.customfields.CustomField;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+
+public class EventContextFieldSupplier implements LogbackContextFieldSupplier {
+
+    @Override
+    public Map<String, Object> map(ILoggingEvent event) {
+        Map<String, Object> result = new HashMap<>();
+        result.putAll(event.getMDCPropertyMap());
+        Object[] arguments = event.getArgumentArray();
+        if (arguments != null) {
+            for (Object argument: arguments) {
+                if (argument instanceof CustomField) {
+                    CustomField customField = (CustomField) argument;
+                    result.put(customField.getKey(), customField.getValue());
```

Comment:
does this still respect the custom field registrations?
as the result is filled with both custom and normal fields alike.

Reply:
This class just provides all fields as String-Object pairs. The decision on the JSON representation with repsect to the configured custom fields is done in the JsonEncoder.